test/Microsoft.Framework.PackageManager.FunctionalTests/DnuRestoreTests.cs

```diff
@@ -189,5 +189,51 @@ public void DnuRestore_ReinstallsCorruptedPackage(string flavor, string os, stri
                 Assert.True(File.Exists(nuspecPath));
             }
         }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DnuRestore_ReinstallsPackageWithNormalizedVersion(string flavor, string os, string architecture)
+        {
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            using (var tempDir = new DisposableDir())
+            {
+                var projectDir = Path.Combine(tempDir, "project");
+                var packagesDir = Path.Combine(tempDir, "packages");
+                var projectJson = Path.Combine(projectDir, Runtime.Project.ProjectFileName);
+
+                Directory.CreateDirectory(projectDir);
+                File.WriteAllText(projectJson, @"
+{
+  ""dependencies"": {
+    ""alpha"": ""0.1.0""
+  }
+}");
+                DnuTestUtils.ExecDnu(
+                    runtimeHomePath,
+                    subcommand: "restore",
+                    arguments: $"{projectDir} -s {_fixture.PackageSource} --packages {packagesDir}");
+
+                // rename package folder to an unnormalized string
+                Directory.Move(Path.Combine(packagesDir, "alpha", "0.1.0"),
```

Comment:
It's actually not possible. Either both normalized or not. https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Framework.PackageManager/Utils/NuGetPackageUtils.cs#L30

But the simplified model used by test is sufficient because the version of a package on disk is parsed from it's folder name: https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Framework.Runtime/NuGet/Repositories/PackageRepository.cs#L102.